src/main/java/net/snowflake/ingest/streaming/SnowflakeStreamingIngestChannel.java

```diff
@@ -240,10 +240,20 @@ public interface SnowflakeStreamingIngestChannel {
    * values.
    *
    * @param rows object data to write
-   * @param offsetToken offset of last row in the row-set, used for replay in case of failures, It
-   *     could be null if you don't plan on replaying or can't replay
+   * @param startOffsetToken start offset of the batch/row-set
+   * @param endOffsetToken end offset of the batch/row-set, used for replay in case of failures, *
+   *     It could be null if you don't plan on replaying or can't replay
    * @return insert response that possibly contains errors because of insertion failures
    */
+  InsertValidationResponse insertRows(
+      Iterable<Map<String, Object>> rows,
+      @Nullable String startOffsetToken,
+      @Nullable String endOffsetToken);
```

Comment:
Is this transparently invoked by the connector? There's nothing user facing here that has to change right?

Reply:
No, this is a new interface and the old one is still there for backward compatibility